specifications/move_adapter/README.md

```diff
@@ -308,86 +327,96 @@ pub enum WriteOp {
-        * [execute](#Script-Execution) script (Move VM)
-        - check the set of events returned and if a configuration change event is published,
-        on return, the adapter marks all transactions not executed yet as `Retry`
-    - TransactionPaylod::Module
-        - checks that module publishing is allowed
-        * [execute](#Function-Execution) `LibraAccount::prologue` (Move VM)
-        - publish module (Move VM)
+Regardless of the kind of transaction, the `WriteSet` changes that it produces are stored in the adapter's data cache, so they will be seen when processing subsequent transactions.
+
+The adapter needs to check if a transaction reconfigures the blockchain. It does that by checking the generated events to see if a `NewEpochEvent` was emitted. If so, the `execute_block` function returns after marking the `TransactionStatus` of all subsequent transactions as `Retry`.
+
+Errors for user transactions, either running scripts or publishing modules, do not stop the execution. A failed user transaction may be kept on-chain to charge for gas, or it may be discarded, but it will not cause the `execute_block` function to return an error. However, if an error occurs for a `WriteSet` transaction or other system transaction, the error is immediately propagated to the result of the `execute_block` function, so that none of the input transactions are executed.
+
+### Script and Module Transactions
+
+After revalidating a user transaction, the adapter processes the payload, depending on its contents. In the common case, the payload is either a script or a module:
+
+  * `Script`: The Move VM is used to [execute](#Script-Execution) the script with the types and arguments specified in the transaction.
+
+  * `Module`: The Move VM is used to [publish](#Publishing) the code module from the transaction.
+
+The Move VM operations used here consume gas according to the gas schedule that is stored in an on-chain configuration. The adapter loads that gas schedule before invoking the VM.
+
+If the script or module payload is processed successfully, the Move VM is next used to [execute](#Function-Execution) the `epilogue` function from the `LibraAccount` module. The epilogue increments the sender's `sequence_number` and deducts the transaction fee based on the gas price and the amount of gas consumed. This function execution is done using the same VM `Session` that was used when processing the payload, so that all the side effects are combined. The epilogue function is run with gas metering disabled.
+
+On the other hand, if an error occurs when processing the payload, the adapter will still run the epilogue function, but it will do so in a new VM `Session` so that any side effects from the payload will not be saved. This has the effect of charging the transaction fee for gas consumption to the sender's account.
+
+If an error occurs while running the epilogue function, then the transaction is discarded.
```

Comment:
is this accurate? I thought the epilogue was error free?

Reply:
Errors are not expected but the epilogue can still in theory fail with `UNEXPECTED_ERROR_FROM_KNOWN_MOVE_FUNCTION`

Reply:
Somehow I also lost the description of the case where the script succeeds but the epilogue fails (e.g., sent payment causes insufficient balance to pay gas). I will fix that, too